gcd/src/main/java/org/spine3/server/storage/datastore/DsRecordStorage.java

```diff
@@ -364,6 +372,77 @@ protected Kind getKind() {
     }
 
     /**
+     * Creates new instance of the {@link Builder}.
+     *
+     * @param <I> the ID type of the instances built by the created {@link Builder}
+     * @return new instance of the {@link Builder}
+     */
+    public static <I> Builder<I> newBuilder() {
+        return new Builder<>();
+    }
+
+    /**
+     * A builder for the {@code DsRecordStorage}.
+     */
+    public static class Builder<I> {
+
+        private Descriptor descriptor;
+        private DatastoreWrapper datastore;
+        private boolean multitenant;
+        private ColumnTypeRegistry<? extends DatastoreColumnType<?, ?>> columnTypeRegistry;
+
+        private Builder() {
+            // Avoid direct initialization
+        }
+
+        /**
+         * @param descriptor the record state {@link Descriptor}
+         */
+        public Builder<I> setDescriptor(Descriptor descriptor) {
```

Comment:
It's a weird mandatory parameter for the public API. Please try to find some way to submit the same information easier; for instance, by requesting the `Message` class instead.